### PR TITLE
Support deltaLog loading on growing segment

### LINKED ISSUE #16885: [Bug]: Growing segments should be able to load delta logs.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: latest master
- Deployment mode(standalone or cluster): N/A
- SDK version(e.g. pymilvus v2.0.0rc2): N/A
- OS(Ubuntu or CentOS): N/A
- CPU/Memory: N/A
- GPU: N/A
- Others:
```

### Current Behavior

Growing segment with valid checkpoints could have delta logs.

When growing segment is being loaded, query node will try to load delta logs for growing segment.

If there's no delta log for the growing segment, we are good. Otherwise, an error will occur and cause the whole load to fail: https://github.com/milvus-io/milvus/blob/2d0f908dbafeb7439a5bbc2e35d9b5726cf33bce/internal/querynode/segment.go#L735-L738

A decent fix would be to fully support loading delta logs while loading growing segments.

To do this, we need to do at least the following:
1. Fully support loading delta logs for growing segments. For example: https://github.com/milvus-io/milvus/blob/2d0f908dbafeb7439a5bbc2e35d9b5726cf33bce/internal/core/src/segcore/segment_c.cpp#L213
2. When 1. is done, remove the limit:
https://github.com/milvus-io/milvus/blob/2d0f908dbafeb7439a5bbc2e35d9b5726cf33bce/internal/querynode/segment.go#L735-L738

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Milvus Log

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by letian-jiang**
Fix #16885 by Implementing LoadDeletedRecord method on segcore growing segment

Signed-off-by: Letian Jiang <letian.jiang@zilliz.com>

### CODE REVIEW

**Comment by letian-jiang** [comment]
/hold 
more tests will be added

File: internal/core/src/segcore/SegmentGrowingImpl.cpp [hunk 1]
```diff
@@ -194,6 +194,28 @@ SegmentGrowingImpl::GetMemoryUsageInBytes() const {
     return total_bytes;
 }
 
+void
+SegmentGrowingImpl::LoadDeletedRecord(const LoadDeletedRecordInfo &info) {
+    AssertInfo(info.row_count > 0, "The row count of deleted record is 0");
+    AssertInfo(info.primary_keys, "Deleted primary keys is null");
+    AssertInfo(info.timestamps, "Deleted timestamps is null");
+    // step 1: get pks and timestamps
+    auto field_id = schema_->get_primary_field_id().value_or(FieldId(-1));
```
Comment (xige-16) [inline internal/core/src/segcore/SegmentGrowingImpl.cpp:203]: INVALID_FIELD_ID is recommended, not -1

Comment (letian-jiang) [reply internal/core/src/segcore/SegmentGrowingImpl.cpp]: Fixed

**Comment by letian-jiang** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**